Tuturial-Structure-Template.md

```diff
@@ -1,68 +1,178 @@
+- Good knowledge of [Multi-factor authentication](https://en.wikipedia.org/wiki/Multi-factor_authentication).
 
-When embedding code using GitHub Gists, if the code is only a snippet make sure to denote that in the file name following the format: `functionName+snippet.fileExtension`  
 
-For example if my function name is generateToken and its a javascript snippet, the Gist would have the title: `generateToken+snippet.js`  
+## Project Setup 
 
-If the code within the Gist is an entire file or set of functions that can be copied and pasted and will work on their own as a library or project, you can just name the Gist after the file they would represent with the repository.  
+> Describe the steps to create a new project and set up the environment.
+> 
+> **IF NONE ARE NEEDED, REMOVE THIS SECTION**
+> 
+> For example:
 
-## Testing XYZ ## 
-> (use as section title; replace XYZ with the project or feature being tested)
+To create the environment necessary to **TITLE OF THIS PAGE**:
 
-Write the steps to test the project.  
+1. Do this.
+1. Do that.
+1. Do something else.
+
+> Markup commands to run in the terminal as ```code```.
+> 
+> If you need a specific file structure, do it as a text diagram:
+> For example: 
+> 
+> <environment_root> \
+> ├── index.html \
+> ├── scripts \
+> │ └── script.js \
+> └── styles \
+> │ └── style.css
+
+## Build \<A rewording of the page title>
+
+> If there are multiple logical sections in the procedure such as back-end and frond-end, use subsections. The goal is to modularize the various elements and build the project in a logical way that makes it easy for the user to follow along.
+> 
+> Write an intro sentence to explain the subsections:
+> 
+> For example:
+
+Step-up authentication implementation involves changes to components in your Agora Client and security infrastructure. This section shows you how to:
+
+- [Build \<module1 in procedure>](link to module1)
+- [Build \<module2 in procedure>](link to module2)
+
+> If there are no locical subsections, write a single procedure here.
+
+### Build \<module1 in procedure> 
+
+> A module explains how to implement a set of functions that operate together to produce a specific feature of the project. 
+> Be sure to explain all inputs and outputs, discuss their significance within the code and how they work together.
+> 
+> All code snippets should use: 
+> - Blog or Medium - GitHub Gists. if the code is only a snippet make sure to denote that in the file name following the format: `functionName+snippet.fileExtension`. For example if my function name is generateToken and its a javascript snippet, the Gist would have the title: `generateToken+snippet.js`. If the code within the Gist is an entire file or set of functions that can be copied and pasted and will work on their own as a library or project, name the Gist after the file they would represent with the repository.
```

Comment:
Not sure what this means

Reply:
It is for DevRel

Reply:
How to denote a code snippet in the file name? Is this only for DevRel?